### PR TITLE
2024 qc fixes

### LINKED ISSUE #42: Get rid of ctd_to_bottle dependency in bokeh tools
`ctd_to_bottle.py` regenerates data that is already created. Let's try and have bokeh use bottle/ctd data instead of needing the script

### LINKED ISSUE #62: Repair QC tool and improve robustness
The QC tool is functional on the DMB and GP17-ANT branches but needs to be repaired on main.

Currently, the 2022 version on Main is not functional and is limited.

- [x] Visualize temperature, salinity, oxygen
- [x] Restore residuals plots for each
- [ ] Change flags to _CTD_ manual flags, as to not overwrite the manual discrete sample flags

### PR DESCRIPTION
**Pull request by Klankers**
Restores enhancements from various branches, including multiple params in the table, new residuals, a residuals plot.

Fixed a few bugs.
* Y-axis defaults to max press + 5% when updated
* X-axis re-centers when updated
* `o_res`,`t_res`,`CTD param`, and `reference` no longer blank out with comment or flag updates

### CODE REVIEW

**Comment by asmith75218** [comment]
The ten files changed I currently see appear to mostly be from PR #123. I'll hold off until that one is merged.

File: ctdcal/tools/data_qc.py
Comment (asmith75218) [inline ctdcal/tools/data_qc.py:1]: This file OK. I think this is the only one that needs review for this PR?

Comment (Klankers) [reply ctdcal/tools/data_qc.py:1]: Yeah. The other changes are a rebase from the changes this afternoon.

**Review by asmith75218 — APPROVED** [review]
Approving changes to data_qc file.

**Pull request merged**
